docs/tutorial.md

```diff
@@ -0,0 +1,436 @@
+
+### Working with Pandas
+
+If you are going to be using the scientific Python library [Pandas](https://pandas.pydata.org/) for any processing or analysis, you may wish use Pandas methods to load your data. This is used to load and manipulate data like we have in our CSV file. Note that for this section we're going to run a very simple computation, the references will have links to more extensive resources for learning more.
+
+```python
+# process_monotremes.py
+
+import pandas
+
+echidna = pandas.read_csv("echidna_minimal.csv")
+platypus = pandas.read_csv("platypus_minimal.csv")
+
+echidna_likes = echidna["public_metrics.like_count"].sum()
+platypus_likes = platypus["public_metrics.like_count"].sum()
+
+print(f"Total likes on echidna tweets: {echidna_likes}. Total likes on platypus tweets: {platypus_likes}.")
+```
+
+Run this script through Python to see which of the monotremes is the coolest:
+
+```shell
+python process_monotremes.py
+```
+
+### Answering the research question: which monotreme is the coolest?
+
+At the time of creating this tutorial, the above script run with the just collected data leads to the following result:
+
+```shell
+Total likes on echidna tweets: 1787652. Total likes on platypus tweets: 3462715.
+```
+
+On that basis, we can conclude that at the time of running this search the platypus is nearly twice as cool as measured by Twitter likes.
+
+Of course this is a simplistic approach to answering this specific research question - we could have made many other choices. Even using a simple quantitative approach looking at metrics: we could have chosen to look at other engagement counts like the number of retweets, or looked at the number of followers of the accounts tweeting about each animal (because a "cooler" account will have more followers). Much of the challenge in using Twitter for research is both about asking the right research question and also the choosing the right approach to the data to address that research question.
+
+## Prepare a dataset for sharing/using a shared dataset
+
+Having performed this analysis and come to a conclusion, it is good practice to share the underlying data so other people can reproduce these results (with some caveats). Noting that we want to preserve Twitter user's agency over the availability of their content, and Twitter's Developer Agreement, we can do this by creating a dataset of tweet IDs. Instead of sharing the content of the tweets, we can share the unique ID for that tweet, which allows others to `hydrate` the tweets by retrieving them again from the Twitter API.
```

Comment:
```suggestion
Having performed this analysis and come to a conclusion, it is good practice to share the underlying data so other people can reproduce these results (with some caveats). Noting that we want to preserve Twitter users' agency over the availability of their content, and Twitter's Developer Agreement, we can do this by creating a dataset of tweet IDs. Instead of sharing the content of the tweets, we can share the unique ID for that tweet, which allows others to `hydrate` the tweets by retrieving them again from the Twitter API.
```